src/voxel.cpp

```diff
@@ -283,21 +288,25 @@ void Voxel::traverseTree
 bool Voxel::isAtom(const Atom& atom, const Vector& pos_vxl, const double rad_vxl, const double rad_probe){
   Vector dist = pos_vxl - atom.getPosVec();
 
-  if((dist < atom.getRad() - rad_vxl) && (0 < atom.getRad() - rad_vxl)){
+  // TODO the second condition would not be necessary if dist was a double type
+  // if(dist + rad_vxl < atom.getRad()){
+  if((dist < atom.getRad() - rad_vxl) && (0 < atom.getRad() - rad_vxl)){ // if completely inside atom
     _type = 0b00000011;
     return true;
   }
-  else if (dist < atom.getRad() + rad_vxl){
+  else if (dist < atom.getRad() + rad_vxl){ // if partially inside atom
     if (readBit(_type,1)){return false;} // if inside atom
     _type = 0b10000010;
   }
-  else if ((dist < atom.getRad() + rad_probe - rad_vxl) && (0 < atom.getRad() + rad_probe - rad_vxl)){
+  // TODO the second condition would not be necessary if dist was a double type
+  // else if (dist + rad_vxl < atom.getRad() + rad_probe){
+  else if ((dist < atom.getRad() + rad_probe - rad_vxl) && (0 < atom.getRad() + rad_probe - rad_vxl)){ // if outside atom but not touching potential probe core
     if (readBit(_type,1)){return false;} // if mixed or inside atom
-    _type = 0b00010000;
+    _type = s_masking_mode? 0b01000000 : 0b00010000;

```

Comment:
Under this condition, the voxels could be either probe shell or excluded void type. What is the reason for giving a temporary unassigned probe shell type?
Wouldn't everything be simpler if unassigned voxels were simply left with type 0b00000000 ?

Reply:
`0b00000000` in this section of the code is used to mark potential core type. Line 201 reads:
`if (_type == 0){_type = s_masking_mode? 0b00100001 : 0b00001001;}`
However, it's true that the condition is not needed. It works just the same if the voxel is assigned any other non '0', non-mixed, non-assigned type (for instance `0b00000010`). However, I've tested whether this improves the runtime and at least for me it makes no noticeable difference. Keeping it as it is makes it more readable for me.

Reply:
The goal is to tell how much time passed between the user pressing the button and getting the result.